### PR TITLE
Run shellcheck and option parsing check on pushes and pull requests

### PR DESCRIPTION
**Pull request by rafasc**
I'm willing to merge a PR that adds the `parseopt-consistency` test.
_Originally posted by @AladW in https://github.com/AladW/aurutils/pull/667#issuecomment-577803145_

I also included a small github action that runs it automatically on pushes and pull requests.
For now, only the parseopt check is done.

Later, we can include other tests and the shellcheck run. I left it out of this PR because I still didn't investigate how to cache/reuse the expensive `pacman -Syu`.

Or I can drop the actions commit for a later PR.

### CODE REVIEW

**Comment by rafasc** [comment]
[](https://github.com/rafasc/aurutils/runs/406090833) as expected as mentioned in #667

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: CI
+on: [push, pull_request]
+jobs:
+  tests:
+    runs-on: ubuntu-latest
```
Comment (AladW) [inline .github/workflows/ci.yml]: Magic 🧙‍♂️

**Comment by AladW** [comment]
I thought `shellcheck` might complain on the unquoted AUR_DEBUG, but looks like it doesn't!

**Comment by AladW** [comment]
> running the test script under an unprivileged user would require we set up one inside the container.

Is that a hard thing to do? I'd never thought `AUR_ASROOT` would be used for something like this 😆

**Comment by Earnestly** [comment]
You can use the `nobody` user for that purpose.  Make sure you create a directory owned by `nobody` to use it.

**Comment by rafasc** [comment]
Added shellcheck.

---
The easiest way to get around the root issue (besides setting  AUR_ASROOT=1) would be to run it directly on the host machine (ubuntu-latest). But we depend on things being sourced from archlinux. c.f `/usr/share/makepkg/util/{parseopt,message,util}.sh`

Alternatively, we could write our own Docker-based github action that sets up the correct environment, including creating the user, setting a repo to be used with other tests, etc.

The problem with this is that the pacman -Syu needs to be cached and I still didn't look into how that can be done. 

Another could be to publish our own archlinux image with the necessary dependencies to the registry. Then change `container: archlinux/base` to our own and remove the `pacman -Syu` step. Then set up something to rebuild it regularly.

To use the `nobody` user we would need to run su -s /bin/bash -c blabla nobody for each step, as they run in a different subshell. We would also run into permission issues because the `nobody` user does not have permission to write in the working directory.

So I went the easy route and disabled the molly-guard.

**Comment by AladW** [comment]
I'm not sure I understand the involved complexity. Compare e.g. https://github.com/AladW/aurutils/blob/ad/next/Dockerfile#L9-L10 - couldn't that replicated in the `.yml` file?

**Comment by rafasc** [comment]
We could do something like this, sure:
``` yaml
    steps:
      - name: Container Setup
        run: pacman -Syu --noconfirm base base-devel shellcheck
      
      - name: Set up unprivileged user
        run:  useradd -m -s /bin/bash archie

      - uses: actions/checkout@v2
      
      - run: chown -R archie $GITHUB_WORKSPACE

      - name: Option parsing consistency
        run: su archie -c test/parseopt-consistency

      - name: Shellcheck
        if: success() || failure() # Do not fail-fast
        run: su archie -c 'make shellcheck'
```
But I do not see the point for this specific job.

What I really dislike is that 50% of the time is spent hitting the mirrors with `run: pacman -Syu --noconfirm base base-devel shellcheck`.

**Comment by AladW** [comment]
Would pointing to a snapshot from `archive.archlinux.org` help?

**Comment by AladW** [comment]
https://github.com/actions/cache 🤔

**Comment by AladW** [comment]
Considering it's already useful in its current state, let's merge this and figure out the details later.

**Comment by rafasc** [comment]
I think we could use `actions/cache` + `archive.archlinux.org`, using the `archive.archlinux.org/repos/week/lastupdate` as the cache key.

Another thing I noticed with issue 672, is that pull requests originating from aladw/aurutils will trigger it twice. Once from the push, and another from the PR.

**Comment by AladW** [comment]
> I think we could use actions/cache + archive.archlinux.org, using the archive.archlinux.org/repos/week/lastupdate as the cache key.

👍 

> Another thing I noticed with issue 672, is that pull requests originating from aladw/aurutils will trigger it twice. Once for the push, and another for the PR.

I noticed the same, wasn't sure if it's normal or not...

**Pull request merged**